salt/docker/remove-pre-reboot.sls

```diff
@@ -0,0 +1,16 @@
+# cleanup all the things we have created
+
+/etc/systemd/system/docker.service.d/proxy.conf:
+  file.absent
+
+/etc/docker/daemon.json:
+  file.absent
+
+/etc/sysconfig/docker:
+  file.absent
+
+/etc/docker/certs.d:
+  file.absent
+
+/var/lib/docker:
```

Comment:
I'm concerned this will not work. If I do a simple `rm -rf`:

```
linux:~ # rm -rf /var/lib/docker
rm: cannot remove '/var/lib/docker/containers/17fd59d6e1760f4c15122240cad7e8f92bd3f6c682c99ffb5982f075004760b1/shm': Device or resource busy
rm: cannot remove '/var/lib/docker/containers/17fd59d6e1760f4c15122240cad7e8f92bd3f6c682c99ffb5982f075004760b1/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/272df0b1060f3307380ac24a267ea6a9697d90070a0e2f9244ba86b51e2af360/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/2bb9f6a9d75edc7eedec06777dfd88693677cf7be25104e6c1b43ca320f279dc/shm': Device or resource busy
rm: cannot remove '/var/lib/docker/containers/2bb9f6a9d75edc7eedec06777dfd88693677cf7be25104e6c1b43ca320f279dc/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/4a6c7cab52db2b820498a27d7a65eb4feff0b4f4330801da62a4334fb44d4b4c/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/259c9fdbbcc526ef7f664e169f59221e8cc2dffe6e92e58b3846dcfc48da5cfc/shm': Device or resource busy
rm: cannot remove '/var/lib/docker/containers/259c9fdbbcc526ef7f664e169f59221e8cc2dffe6e92e58b3846dcfc48da5cfc/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/c948344d1b80e4b26c67b07e647a0a25599ed4fde93c51c98e44b8aded802641/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/5e416321998e526c825af89d31eabfda163b053dba79c7d75b878b28da5f177a/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/e894f59666490e30db32566727fba33517a0de5f906975ba754ffb79aa8d747e/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/aa251c688e65ef35f7971f6849c152b0c616edd5aed5cb71d2ba22fce5126d08/shm': Device or resource busy
rm: cannot remove '/var/lib/docker/containers/aa251c688e65ef35f7971f6849c152b0c616edd5aed5cb71d2ba22fce5126d08/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/b307b2a56b457af5b6f8382a211d88de9b399abca245897ae8c52e6f5ccdfa95/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/ca5aa0a86d9a8eec4b64f5640ab35f7a313ae9164ac4c2926a702970b9738ca5/shm': Device or resource busy
rm: cannot remove '/var/lib/docker/containers/ca5aa0a86d9a8eec4b64f5640ab35f7a313ae9164ac4c2926a702970b9738ca5/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/containers/e776b5216e5b1a4e81d0e799c08980538c0cfb22831e2399acb6f7fba96a089c/secrets/suse_286eab516f18b9064e5ff6e7f3888a60032d608714777b0a2e464a25aa8299b7_credentials.d': Read-only file system
rm: cannot remove '/var/lib/docker/plugins': Device or resource busy
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/cba627b25d8dd1fa61c3030079eed15934011e6343de9d56fdd616e92c66c4f0': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/ab48fadae9b000ed092adafb6cb07a1c7cd5cec77f2fd32d09221bbe331a88f1': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/487fe1fc221e6b19ea230dd6a69bba4c7b3e879f9914abd283364bbc81c90f7b': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/233f9675c39382ad1f8dacb300942f8694abd82cf25b48da2abb4c1db4a6040b': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/52e853910f8376952c6dfa3c97c529bab3621ab831df9f51b6497b787d6fb7a8': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/ca356762811ff2af0bd4704439414c270c2501c7ac41fcf81d169d7283926446': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/da83fc0473e44a850ecb51aeda952b4e8a913afe1bffe2b2d2bdb52211290890': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/4e546b9af6804efd752a333c08829fda50468c8954791bbcd789adfcc244eae8': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/669901abfc4a2207259b2aff8f9a7afa8e746a68c15064c8e903d367302f0003': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/cba449733a0118230359a6f1d2c60b3c8bd44da94a21a0b7ba470fa201841202': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/62e5b433e4793100a4ad7536f7fa3a0fcff9ed439b327dc4082c9a16b7e1ee7f': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/ee6f5269daf9af6f4cb056837a2de0786e38fd3be134f8147e9e07f703fd3ba8': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/d12938fd93ae15262751186086fc1220748419a06a8e7cd90537fd16fb12e63e': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/d51f835615d45828857b728d48f8562e74b9b811b2aea887676cf5997f74ce67': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/d887814fd77568ea0c1f899fb8b1895861cffb156493a0bc43373ef65ac7b6b9': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/934930b600affc9f45667edcd6e94e98ce7fd43c0f7bdfb1db3c8b0cb047fa7b': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/f201114f9448598b0018aec07aec6bb65874e5719d458eb29791b4cca9e95b6b-init': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/f201114f9448598b0018aec07aec6bb65874e5719d458eb29791b4cca9e95b6b': Operation not permitted
rm: cannot remove '/var/lib/docker/btrfs/subvolumes/d5805f2b8eeb7aa93a5cfff122c7cf4cfb0eaedc09af22996a33c2db0c097e68-init': Operation not permitted
```

Reply:
Yes, before this will be run, all services will be stopped 
https://github.com/kubic-project/salt/pull/351/files#diff-286f8a76a1c2def34f37383c9b5ac34cR108

Reply:
You are right, hmmm I have to check what will be the best solution for that